api/v1/server/handlers/v1/webhooks/receive.go

```diff
@@ -261,17 +322,57 @@ type IsChallenge bool
 func (w *V1WebhooksService) performChallenge(webhookPayload []byte, webhook sqlcv1.V1IncomingWebhook, request http.Request) (IsChallenge, map[string]interface{}, error) {
 	switch webhook.SourceName {
 	case sqlcv1.V1IncomingWebhookSourceNameSLACK:
-		payload := make(map[string]interface{})
-		err := json.Unmarshal(webhookPayload, &payload)
+		/* Slack sends requests in different formats:
+		 * - Events API URL verification: application/json with direct JSON payload
+		 * - Interactive components: application/x-www-form-urlencoded with a payload field containing JSON
+		 * See: https://docs.slack.dev/interactivity/handling-user-interaction/#payloads
+		 */
+		contentType := request.Header.Get("Content-Type")
+
+		var payloadJSON []byte
+
+		if strings.Contains(contentType, "application/x-www-form-urlencoded") {
+			/* Parse form-encoded data */
+			formData, err := url.ParseQuery(string(webhookPayload))
+			if err != nil {
+				/* If we can't parse form data, it's likely not a challenge - let normal processing handle it */
+				return false, nil, nil
+			}
 
+			/* Extract the payload field which contains JSON */
+			payloadValue := formData.Get("payload")
+			if payloadValue == "" {
+				/* If no payload field, this is not a challenge - let normal processing handle it */
+				return false, nil, nil
+			}
+			payloadJSON = []byte(payloadValue)
+		} else {
```

Comment:
do we need the full payload for the slack challenge? or can we leave this method as-is and only modify the one that actually processes the request?